### PR TITLE
[#177116784] Add Redis config

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes
- Add Redis configuration and client
- Update dependencies and typings

#### Motivation and Context
Since CGN must provide a One Time Password, to be used through online purchase, we must store this temporary data in a proper cache. This will be used also for other optimizations.

#### How Has This Been Tested?
No tests needed.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: utils/config.ts [hunk 1]
```diff
@@ -9,25 +9,40 @@ import * as t from "io-ts";
 import { readableReport } from "italia-ts-commons/lib/reporters";
 import { NonEmptyString } from "italia-ts-commons/lib/strings";
 
+export const RedisParams = t.intersection([
+  t.interface({
+    REDIS_URL: NonEmptyString
+  }),
+  t.partial({
+    REDIS_CLUSTER_ENABLED: t.boolean,
```
Comment (BurnedMarshal) [inline utils/config.ts:17]: If we have this tyings, we need to remap the value in [here](https://github.com/pagopa/io-functions-cgn/pull/30/files#diff-8f4625a1cf4cb245a2f9e780a9a17ff6a7132b8e070997144a67499d7ff793b4R48)
like
```typescript
const errorOrConfig: t.Validation<IConfig> = IConfig.decode({
  ...process.env,
  isProduction: process.env.NODE_ENV === "production",
  REDIS_CLUSTER_ENABLED: fromNullable(process.env.REDIS_CLUSTER_ENABLED)
    .map(_ => _.toLowerCase() === "true")
    .toUndefined()
});
```

File: utils/redis.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        )
+      )
+      .getOrElseL(() =>
+        createSimpleRedisClient(
+          config.REDIS_URL,
+          config.REDIS_PASSWORD,
+          config.REDIS_PORT,
+          true
+        )
+      );
```
Comment (BurnedMarshal) [inline utils/redis.ts:66]: If you like it we can remove the if and use `fromPredicate` instead.
```suggestion
export const REDIS_CLIENT = fromPredicate<boolean>(_ => _)(config.isProduction)
  .mapNullable(_ => config.REDIS_CLUSTER_ENABLED)
  .chain(fromPredicate(_ => _))
  .map(() =>
    createClusterRedisClient(
      config.REDIS_URL,
      config.REDIS_PASSWORD,
      config.REDIS_PORT
    )
  )
  .getOrElseL(() =>
    createSimpleRedisClient(
      config.REDIS_URL,
      config.REDIS_PASSWORD,
      config.REDIS_PORT,
      true
    )
  );
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: utils/config.ts [hunk 3]
```diff
@@ -5,35 +5,54 @@
...
+    REDIS_PASSWORD: NonEmptyString,
+    REDIS_PORT: NonEmptyString
```
Comment (BurnedMarshal) [inline utils/config.ts:20]: Could you add this ENVs inside the `env.example` file? We should upgrade even `io-mock` configuration for `fn-cgn`

**Review by BurnedMarshal — APPROVED** [review]
LGTM apart the missing envs into`env.example`

**Comment by AleDore** [comment]
> Could you add this ENVs inside the `env.example` file? We should upgrade even `io-mock` configuration for `fn-cgn` 

Done -> [https://github.com/pagopa/io-mock/pull/31](https://github.com/pagopa/io-mock/pull/31) :)

**Pull request merged**